### PR TITLE
Rp/fix multipart upload

### PR DESCRIPTION
**Pull request by richpryce**
## What does this change?

Updates the awssdk to fix error when uploading very large files to S3.

### CODE REVIEW

File: common/src/main/scala/com/gu/multimedia/storagetier/framework/MessageProcessingFramework.scala [hunk 1]
```diff
@@ -211,6 +210,7 @@ class MessageProcessingFramework (ingest_queue_name:String,
       .contentType("application/json")
       .contentEncoding("UTF-8")
       .messageId(UUID.randomUUID().toString)
+      .deliveryMode(2)
```
Comment (fredex42) [inline common/src/main/scala/com/gu/multimedia/storagetier/framework/MessageProcessingFramework.scala:213]: does this not belong in another PR?

**Review by fredex42 — APPROVED** [review]
Yeah I am happy with this 😄  Only small gripe - `deliveryMode(2)` belongs in another PR really, but hey let's get this merged and get on with fixing stuff

File: online_archive/src/main/scala/FileUploader.scala [hunk 2]
```diff
@@ -1,27 +1,30 @@
...
+import software.amazon.awssdk.transfer.s3.model.UploadRequest
 
-import java.io.{File, InputStream}
+import java.io.{File, FileInputStream}
 import java.nio.file.{Files, Paths}
+import java.security.{DigestInputStream, MessageDigest}
 import java.util.Base64
-
 import scala.concurrent.{ExecutionContext, Future}
-import scala.util.{Failure, Success, Try}
```
Comment (Dala0) [inline online_archive/src/main/scala/FileUploader.scala:19]: Seeming pointless removable to add the same line again later.

File: online_archive/src/main/scala/FileUploader.scala [hunk 3]
```diff
@@ -1,27 +1,30 @@
...
 import org.apache.commons.codec.binary.Hex
 import org.slf4j.{LoggerFactory, MDC}
 import software.amazon.awssdk.core.async.AsyncRequestBody
 import software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient
 import software.amazon.awssdk.regions.Region
 import software.amazon.awssdk.services.s3.S3Client
-import software.amazon.awssdk.services.s3.model.{HeadObjectRequest, HeadObjectResponse, ListObjectVersionsRequest, NoSuchKeyException, PutObjectRequest}
-import software.amazon.awssdk.transfer.s3.{CompletedUpload, S3ClientConfiguration, S3TransferManager, UploadRequest}
+import software.amazon.awssdk.services.s3.internal.crt.S3CrtAsyncClient
+import software.amazon.awssdk.services.s3.model._
```
Comment (Dala0) [inline online_archive/src/main/scala/FileUploader.scala:12]: This should ideally be kept as it was unless we know only the objects originally imported exist.

File: online_archive/src/test/scala/FileUploaderSpec.scala [hunk 4]
```diff
@@ -1,18 +1,25 @@
 import org.specs2.mock.Mockito
 import org.specs2.mutable.Specification
+import software.amazon.awssdk.core.async.AsyncRequestBody
 import software.amazon.awssdk.services.s3.S3Client
-import software.amazon.awssdk.services.s3.model.{HeadObjectRequest, HeadObjectResponse, ListObjectVersionsRequest, ListObjectVersionsResponse, NoSuchKeyException, ObjectVersion, PutObjectResponse, S3Exception}
-import software.amazon.awssdk.transfer.s3.{CompletedUpload, S3TransferManager, Upload, UploadRequest}
+import software.amazon.awssdk.services.s3.model._
```
Comment (Dala0) [inline online_archive/src/test/scala/FileUploaderSpec.scala:5]: This should ideally be kept as it was unless we know only the objects originally imported exist.

File: online_archive/src/test/scala/FileUploaderSpec.scala [hunk 5]
```diff
@@ -1,18 +1,25 @@
...
-import scala.concurrent.duration._
-import scala.concurrent.ExecutionContext.Implicits.global
 import java.io.File
+import java.nio.charset.StandardCharsets
+import java.security.MessageDigest
+import java.util.Base64
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
 import scala.concurrent.{Await, Future}
-import scala.util.Try
```
Comment (Dala0) [inline online_archive/src/test/scala/FileUploaderSpec.scala:12]: Seeming pointless removal to add the same line again later.

File: online_archive/src/test/scala/FileUploaderSpec.scala [hunk 6]
```diff
@@ -1,18 +1,25 @@
...
-import software.amazon.awssdk.services.s3.model.{HeadObjectRequest, HeadObjectResponse, ListObjectVersionsRequest, ListObjectVersionsResponse, NoSuchKeyException, ObjectVersion, PutObjectResponse, S3Exception}
-import software.amazon.awssdk.transfer.s3.{CompletedUpload, S3TransferManager, Upload, UploadRequest}
+import software.amazon.awssdk.services.s3.model._
+import software.amazon.awssdk.transfer.s3.S3TransferManager
+import software.amazon.awssdk.transfer.s3.model.{CompletedUpload, Upload, UploadRequest}
+import scala.util.Success
 
-import scala.jdk.CollectionConverters._
-import scala.concurrent.duration._
-import scala.concurrent.ExecutionContext.Implicits.global
```
Comment (Dala0) [inline online_archive/src/test/scala/FileUploaderSpec.scala:9]: Seeming pointless removal to add the same line again later.

File: online_archive/src/test/scala/FileUploaderSpec.scala [hunk 7]
```diff
@@ -1,18 +1,25 @@
...
 import software.amazon.awssdk.services.s3.S3Client
-import software.amazon.awssdk.services.s3.model.{HeadObjectRequest, HeadObjectResponse, ListObjectVersionsRequest, ListObjectVersionsResponse, NoSuchKeyException, ObjectVersion, PutObjectResponse, S3Exception}
-import software.amazon.awssdk.transfer.s3.{CompletedUpload, S3TransferManager, Upload, UploadRequest}
+import software.amazon.awssdk.services.s3.model._
+import software.amazon.awssdk.transfer.s3.S3TransferManager
+import software.amazon.awssdk.transfer.s3.model.{CompletedUpload, Upload, UploadRequest}
+import scala.util.Success
 
-import scala.jdk.CollectionConverters._
-import scala.concurrent.duration._
```
Comment (Dala0) [inline online_archive/src/test/scala/FileUploaderSpec.scala:8]: Seeming pointless removal to add the same line again later.

File: online_archive/src/test/scala/FileUploaderSpec.scala [hunk 8]
```diff
@@ -1,18 +1,25 @@
...
+import software.amazon.awssdk.core.async.AsyncRequestBody
 import software.amazon.awssdk.services.s3.S3Client
-import software.amazon.awssdk.services.s3.model.{HeadObjectRequest, HeadObjectResponse, ListObjectVersionsRequest, ListObjectVersionsResponse, NoSuchKeyException, ObjectVersion, PutObjectResponse, S3Exception}
-import software.amazon.awssdk.transfer.s3.{CompletedUpload, S3TransferManager, Upload, UploadRequest}
+import software.amazon.awssdk.services.s3.model._
+import software.amazon.awssdk.transfer.s3.S3TransferManager
+import software.amazon.awssdk.transfer.s3.model.{CompletedUpload, Upload, UploadRequest}
+import scala.util.Success
 
-import scala.jdk.CollectionConverters._
```
Comment (Dala0) [inline online_archive/src/test/scala/FileUploaderSpec.scala:7]: Seeming pointless removal to add the same line again later.

**Review by Dala0 — APPROVED** [review]
